sound/soc/sof/trace.c

```diff
@@ -81,6 +89,12 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 		return -EIO;
 	}
 
+	/* end-of-file reached, force position to zero */
+	if (avail == 0) {
```

Comment:
@kv2019i Is this whole if block needed, if avail is 0, count below will be 0 and we will copy nothing and return 0 anyway isnt it?

Reply:
@ranj063 No, you are right. The code would work by just setting the EOF flag here and continue code.  I ended up putting return directly here as it seemed more readable (as we have direct returns elsewhere in the functin as well).

Reply:
@kv2019i why will we need to set eof here? would we reach here only if the trace_sleep is woken up and eof is already set in the sof_wait_trace_avail()?

Reply:
@ranj063 wrote:
> @kv2019i why will we need to set eof here? would we reach here only if the trace_sleep is woken up and eof is already set in the sof_wait_trace_avail()?

There are two cases how we hit eof:
1) trace client sleeping in read -> runtime suspend -> eof for trace with above code
2) trace client not sleeping -> runtime suspend -> eof signalled when all pending data is read by trace app 

Hmm, I can think whether this could be done in one place, but for now I had to have the eof marking in two places to cover all cases.